Automattic/wp-calypso
# NUX: Update/pass new user signup data to happy chat

## Changes proposed in this Pull Request

The purpose of this PR is to give Happy Chat users more context on new users, especially those who are currently attempting our signup process, by adding the following information to the [data we send to Happy Chat](https://github.com/Automattic/wp-calypso/blob/f32fd2d/client/me/help/help-contact/index.jsx#L119):

```
currentSignupStatus = {
    flowName: 'current flow name',
    currentStep:  'current step name',
    lastIncompleteStep: 'the last step the user tried but did not complete',
};
```
❓I'd like some feedback on the property names/values! :)

## Next Steps

This PR only serves to collect and package the data. The next task is to allow the Happy Chat application to consume the new properties. 

## Testing instructions

One way to manually test is to ensure that the new properties are being passed to the `sendUserInfo` [action](https://github.com/Automattic/wp-calypso/blob/f32fd2d/client/state/happychat/connection/actions.js#L260) in the `info` argument. 

The low-pain way to do this is add `variationSlug = SUPPORT_HAPPYCHAT;` before the [switch statement](https://github.com/Automattic/wp-calypso/blob/f32fd2d/client/me/help/help-contact/index.jsx#L288), to force the Happy Chat form. 

_Extra points for returning early_ in the [action](https://github.com/Automattic/wp-calypso/blob/f32fd2d/client/state/happychat/connection/actions.js#L260) so you don't trigger a chat window.

1. Switch on debugging: `localStorage.setItem( 'debug', 'calypso:*' );`
2. In a logged-in state, enter the signup process http://calypso.localhost:3000/start
3. Complete a couple of steps but not all of them
4. Return to your dash, open the inline help popup, and click **Chat with us** (

Check debug for `Starting HappyChat with the following user information` and the packet of data that is sent off when a user initiates a Happy Chat. 

.
